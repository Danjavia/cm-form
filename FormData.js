import React, { Component } from 'react';
import { Form } from 'antd';
import FormEntry from './FormEntry';

/**
 * FormData class
 * Extend class for working with antd forms
 * */
export default class FormData extends Component {

  /**
   * render
   * @return {MarkupHTML} Html rendered
   * */
  render() {
    const { form, questions } = this.props;
    const formItemProperties = {
      labelCol: { span: 23 },
      wrapperCol: { span: 23 },
    };

    return (
      <div>
        {this.props.formTitle && <h2 style={{marginLeft: 10}}>
          {this.props.formTitle || 'Datos generales'}
        </h2>}

        <Form layout="vertical" onSubmit={this.props.onSubmit}>
          <div style={{ position: 'relative' }}>
            {questions &&
              questions.map(
                (props, key) =>
                  props.visible !== false &&
                    <section key={key}>
                      <FormEntry form={form} key={key} formItemProperties={formItemProperties} {...props} />

                    {props.children &&
                      props.children.map(
                        (props, key) =>
                          props.visible && <section key={key}>
                            <FormEntry form={form} key={key} formItemProperties={formItemProperties} {...props} />

                            {props.children &&
                              props.children.map((props, key) =>
                                props.visible && <FormEntry form={form} key={key} formItemProperties={formItemProperties} {...props} />
                              )}
                          </section>,
                      )}
                  </section>,
              )}

            {this.props.submitText && <FormEntry
              form={form}
              formEntryType="submitButton"
              buttonText={this.props.submitText}
              type={this.props.submitButtonType || 'primary'}
              disabled={this.props.disabled || false}
              />}
            {this.props.children}
          </div>
        </Form>
      </div>
    );
  }
}
