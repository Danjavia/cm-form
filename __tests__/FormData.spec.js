import React from 'react';
import { shallow } from 'enzyme';
import FormData from '../FormData';

const data = {
  firstName: 'Danny',
  secondName: 'Hoower',
  firstSurename: 'Viasus',
  secondSurename: 'Avila',
  emailAddress: 'email@exmaple.com',
  mobilePhone: '573227808655',
};

const questions = data => [
  {
    formEntryType: 'input',
    fieldName: 'firstName',
    initialValue: `${data.firstName} ${data.secondName}`,
    inputProps: {
      disabled: true,
    },
  },
  {
    formEntryType: 'select',
    fieldName: 'city',
    options: [
      {
        name: 'Bogota', value: 'bogota'
      },
      {
        name: 'Tunja', value: 'tunja',
      },
      {
        name: 'vali', value: 'cali'
      }
    ]
  },
  {
    formEntryType: 'datepicker',
    fieldName: 'datetime',
  },
  {
    formEntryType: 'radioGroup',
    fieldName: 'phoneMobile',
    options: [
      {
        name: 'Bogota', value: 'bogota'
      },
      {
        name: 'Tunja', value: 'tunja',
      },
      {
        name: 'vali', value: 'cali'
      }
    ]
  },
  {
    formEntryType: 'radioGroup',
    fieldName: 'phoneMobile',
    vertical: true,
    options: [
      {
        name: 'Bogota', value: 'bogota'
      },
      {
        name: 'Tunja', value: 'tunja',
      },
      {
        name: 'vali', value: 'cali'
      }
    ]
  },
];


describe('FormData', () => {
  let component, props;

  beforeEach(() => {
    props = {
      form: {},
      questions: questions(data),
    };
    component = shallow(<FormData {...props} />);
  });

  test('screenshot test', () => {
    expect(component).toMatchSnapshot();
  });
});
