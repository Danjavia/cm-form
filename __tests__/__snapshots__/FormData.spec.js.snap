// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormData screenshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Form
      hideRequiredMark={false}
      layout="vertical"
      onSubmit={[Function]}
      prefixCls="ant-form"
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <section>
          <FormEntry
            fieldName="firstName"
            form={Object {}}
            formEntryType="input"
            formItemProperties={
              Object {
                "labelCol": Object {
                  "span": 23,
                },
                "wrapperCol": Object {
                  "span": 23,
                },
              }
            }
            initialValue="Danny Hoower"
            inputProps={
              Object {
                "disabled": true,
              }
            }
          />
        </section>
        <section>
          <FormEntry
            fieldName="city"
            form={Object {}}
            formEntryType="select"
            formItemProperties={
              Object {
                "labelCol": Object {
                  "span": 23,
                },
                "wrapperCol": Object {
                  "span": 23,
                },
              }
            }
            options={
              Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ]
            }
          />
        </section>
        <section>
          <FormEntry
            fieldName="datetime"
            form={Object {}}
            formEntryType="datepicker"
            formItemProperties={
              Object {
                "labelCol": Object {
                  "span": 23,
                },
                "wrapperCol": Object {
                  "span": 23,
                },
              }
            }
          />
        </section>
        <section>
          <FormEntry
            fieldName="phoneMobile"
            form={Object {}}
            formEntryType="radioGroup"
            formItemProperties={
              Object {
                "labelCol": Object {
                  "span": 23,
                },
                "wrapperCol": Object {
                  "span": 23,
                },
              }
            }
            options={
              Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ]
            }
          />
        </section>
        <section>
          <FormEntry
            fieldName="phoneMobile"
            form={Object {}}
            formEntryType="radioGroup"
            formItemProperties={
              Object {
                "labelCol": Object {
                  "span": 23,
                },
                "wrapperCol": Object {
                  "span": 23,
                },
              }
            }
            options={
              Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ]
            }
            vertical={true}
          />
        </section>
      </div>
    </Form>
  </div>,
  "nodes": Array [
    <div>
      <Form
        hideRequiredMark={false}
        layout="vertical"
        onSubmit={[Function]}
        prefixCls="ant-form"
      >
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <section>
            <FormEntry
              fieldName="firstName"
              form={Object {}}
              formEntryType="input"
              formItemProperties={
                Object {
                  "labelCol": Object {
                    "span": 23,
                  },
                  "wrapperCol": Object {
                    "span": 23,
                  },
                }
              }
              initialValue="Danny Hoower"
              inputProps={
                Object {
                  "disabled": true,
                }
              }
            />
          </section>
          <section>
            <FormEntry
              fieldName="city"
              form={Object {}}
              formEntryType="select"
              formItemProperties={
                Object {
                  "labelCol": Object {
                    "span": 23,
                  },
                  "wrapperCol": Object {
                    "span": 23,
                  },
                }
              }
              options={
                Array [
                  Object {
                    "name": "Bogota",
                    "value": "bogota",
                  },
                  Object {
                    "name": "Tunja",
                    "value": "tunja",
                  },
                  Object {
                    "name": "vali",
                    "value": "cali",
                  },
                ]
              }
            />
          </section>
          <section>
            <FormEntry
              fieldName="datetime"
              form={Object {}}
              formEntryType="datepicker"
              formItemProperties={
                Object {
                  "labelCol": Object {
                    "span": 23,
                  },
                  "wrapperCol": Object {
                    "span": 23,
                  },
                }
              }
            />
          </section>
          <section>
            <FormEntry
              fieldName="phoneMobile"
              form={Object {}}
              formEntryType="radioGroup"
              formItemProperties={
                Object {
                  "labelCol": Object {
                    "span": 23,
                  },
                  "wrapperCol": Object {
                    "span": 23,
                  },
                }
              }
              options={
                Array [
                  Object {
                    "name": "Bogota",
                    "value": "bogota",
                  },
                  Object {
                    "name": "Tunja",
                    "value": "tunja",
                  },
                  Object {
                    "name": "vali",
                    "value": "cali",
                  },
                ]
              }
            />
          </section>
          <section>
            <FormEntry
              fieldName="phoneMobile"
              form={Object {}}
              formEntryType="radioGroup"
              formItemProperties={
                Object {
                  "labelCol": Object {
                    "span": 23,
                  },
                  "wrapperCol": Object {
                    "span": 23,
                  },
                }
              }
              options={
                Array [
                  Object {
                    "name": "Bogota",
                    "value": "bogota",
                  },
                  Object {
                    "name": "Tunja",
                    "value": "tunja",
                  },
                  Object {
                    "name": "vali",
                    "value": "cali",
                  },
                ]
              }
              vertical={true}
            />
          </section>
        </div>
      </Form>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FormData
        form={Object {}}
        questions={
          Array [
            Object {
              "fieldName": "firstName",
              "formEntryType": "input",
              "initialValue": "Danny Hoower",
              "inputProps": Object {
                "disabled": true,
              },
            },
            Object {
              "fieldName": "city",
              "formEntryType": "select",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
            },
            Object {
              "fieldName": "datetime",
              "formEntryType": "datepicker",
            },
            Object {
              "fieldName": "phoneMobile",
              "formEntryType": "radioGroup",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
            },
            Object {
              "fieldName": "phoneMobile",
              "formEntryType": "radioGroup",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
              "vertical": true,
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FormData {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "form": Object {},
          "questions": Array [
            Object {
              "fieldName": "firstName",
              "formEntryType": "input",
              "initialValue": "Danny Hoower",
              "inputProps": Object {
                "disabled": true,
              },
            },
            Object {
              "fieldName": "city",
              "formEntryType": "select",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
            },
            Object {
              "fieldName": "datetime",
              "formEntryType": "datepicker",
            },
            Object {
              "fieldName": "phoneMobile",
              "formEntryType": "radioGroup",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
            },
            Object {
              "fieldName": "phoneMobile",
              "formEntryType": "radioGroup",
              "options": Array [
                Object {
                  "name": "Bogota",
                  "value": "bogota",
                },
                Object {
                  "name": "Tunja",
                  "value": "tunja",
                },
                Object {
                  "name": "vali",
                  "value": "cali",
                },
              ],
              "vertical": true,
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Form
            hideRequiredMark={false}
            layout="vertical"
            onSubmit={[Function]}
            prefixCls="ant-form"
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <section>
                <FormEntry
                  fieldName="firstName"
                  form={Object {}}
                  formEntryType="input"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  initialValue="Danny Hoower"
                  inputProps={
                    Object {
                      "disabled": true,
                    }
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="city"
                  form={Object {}}
                  formEntryType="select"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="datetime"
                  form={Object {}}
                  formEntryType="datepicker"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="phoneMobile"
                  form={Object {}}
                  formEntryType="radioGroup"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="phoneMobile"
                  form={Object {}}
                  formEntryType="radioGroup"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                  vertical={true}
                />
              </section>
            </div>
          </Form>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Form
            hideRequiredMark={false}
            layout="vertical"
            onSubmit={[Function]}
            prefixCls="ant-form"
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <section>
                <FormEntry
                  fieldName="firstName"
                  form={Object {}}
                  formEntryType="input"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  initialValue="Danny Hoower"
                  inputProps={
                    Object {
                      "disabled": true,
                    }
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="city"
                  form={Object {}}
                  formEntryType="select"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="datetime"
                  form={Object {}}
                  formEntryType="datepicker"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="phoneMobile"
                  form={Object {}}
                  formEntryType="radioGroup"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                />
              </section>
              <section>
                <FormEntry
                  fieldName="phoneMobile"
                  form={Object {}}
                  formEntryType="radioGroup"
                  formItemProperties={
                    Object {
                      "labelCol": Object {
                        "span": 23,
                      },
                      "wrapperCol": Object {
                        "span": 23,
                      },
                    }
                  }
                  options={
                    Array [
                      Object {
                        "name": "Bogota",
                        "value": "bogota",
                      },
                      Object {
                        "name": "Tunja",
                        "value": "tunja",
                      },
                      Object {
                        "name": "vali",
                        "value": "cali",
                      },
                    ]
                  }
                  vertical={true}
                />
              </section>
            </div>
          </Form>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FormData
    form={Object {}}
    questions={
      Array [
        Object {
          "fieldName": "firstName",
          "formEntryType": "input",
          "initialValue": "Danny Hoower",
          "inputProps": Object {
            "disabled": true,
          },
        },
        Object {
          "fieldName": "city",
          "formEntryType": "select",
          "options": Array [
            Object {
              "name": "Bogota",
              "value": "bogota",
            },
            Object {
              "name": "Tunja",
              "value": "tunja",
            },
            Object {
              "name": "vali",
              "value": "cali",
            },
          ],
        },
        Object {
          "fieldName": "datetime",
          "formEntryType": "datepicker",
        },
        Object {
          "fieldName": "phoneMobile",
          "formEntryType": "radioGroup",
          "options": Array [
            Object {
              "name": "Bogota",
              "value": "bogota",
            },
            Object {
              "name": "Tunja",
              "value": "tunja",
            },
            Object {
              "name": "vali",
              "value": "cali",
            },
          ],
        },
        Object {
          "fieldName": "phoneMobile",
          "formEntryType": "radioGroup",
          "options": Array [
            Object {
              "name": "Bogota",
              "value": "bogota",
            },
            Object {
              "name": "Tunja",
              "value": "tunja",
            },
            Object {
              "name": "vali",
              "value": "cali",
            },
          ],
          "vertical": true,
        },
      ]
    }
  />,
}
`;
